ctapipe/reco/FitGammaHillas.py

```diff
@@ -0,0 +1,308 @@
+                direction vector of the gamma's origin used as seed
+            circles : GreatCircle array
+                collection of great circles created from the camera images
+            weights : array
+                list of weights for each image/great circle
+                
+            Returns:
+            --------
+            MEstimator : float
+                
+                
+            Algorithm:
+            ----------
+            M-Est = sum[  weight * sqrt( 2 * chi**2 ) ]
+            
+            
+            Note:
+            -----
+            seemingly inferior to negative sum of angles...
+            
+        """
+
+        sin_ang = np.array([linalg.length(np.cross(origin,circ.norm)) for circ in self.circles.values()])
+        return sum( weights*np.sqrt( 2.+ (sin_ang-np.pi/2.)**2) )
+        
+        
+        ang = np.array([linalg.angle(origin,circ.norm) for circ in self.circles.values()])
+        ang[ang>np.pi/2.] = np.pi-ang[ang>np.pi/2]
+        return sum( weights*np.sqrt( 2.+ (ang-np.pi/2.)**2) )
+    
+    def _n_angle_sum(self, origin, weights):
+        """ calculates the negative sum of the angle between the fit direction 
+            and all the normal vectors of the great circles
+            
+            Parameters:
+            -----------
+            origin : length-3 array
+                direction vector of the gamma's origin used as seed
+            circles : GreatCircle array
+                collection of great circles created from the camera images
+            weights : array
+                list of weights for each image/great circle
+                
+            Returns:
+            --------
+            n_sum_angles : float
+                negative of the sum of the angles between the test direction
+                and all normal vectors of the given great circles
+        """
+        #sin_ang = np.array([np.dot(origin,circ.norm) for circ in self.circles.values()])
+        sin_ang = np.array([linalg.length(np.cross(origin,circ.norm)) for circ in self.circles.values()])
+        return -sum(weights*sin_ang)
+    
+    
+    def fit_core(self, seed=[0,0]*u.m, test_function=None):
+        if test_function == None: test_function = self._dist_to_traces
+        zdir = np.array([0,0,1])
+        
+        # the core of the shower lies on the cross section of the great circle with the horizontal plane
+        # the direction of this cross section is the cross-product of the normal vectors of the circle and the horizontal plane
+        # here we only care about the direction; not the orientation...
+        for circle in self.circles.values():
+            circle.trace = linalg.normalise( np.cross( circle.norm, zdir) )
+        
+        
+        # minimising the test function
+        self.fit_result_core = minimize( test_function, seed,
+                                         method='BFGS', options={'disp': False}
+                                       )
```

Comment:
Even if the fit fails, the result is not worse than the seed, so I will leave it like this and simply return the result. For people interested in the success status, I set the result as a instance member any way.